test/functional/p2p_invalid_tx.py

```diff
@@ -143,9 +143,6 @@ def run_test(self):
                 "disconnecting peer=0",
         ]):
             node.p2p.send_txs_and_test([tx1], node, success=False, expect_disconnect=True)
-        # send_txs_and_test will have waited for disconnect, so we can safely check that no reject has been received
```

Comment:
This whole sub-test can be removed (everything from `# restart node with sending BIP61 messages disabled, check that it disconnects without sending the reject message` onward)

You could also update L71 higher up to test the reject reason:

```
+++ b/test/functional/p2p_invalid_tx.py
@@ -68,7 +68,8 @@ class InvalidTxRequestTest(BitcoinTestFramework):
         # and we get disconnected immediately
         self.log.info('Test a transaction that is rejected')
         tx1 = create_tx_with_script(block1.vtx[0], 0, script_sig=b'\x64' * 35, amount=50 * COIN - 12000)
-        node.p2p.send_txs_and_test([tx1], node, success=False, expect_disconnect=True)
+        node.p2p.send_txs_and_test([tx1], node, success=False, expect_disconnect=True, 
+                                   reject_reason="{} from peer=0 was not accepted: mandatory-script-verify-flag-failed (Invalid OP_IF construction) (code 16)".format(tx1.hash))
```